website/projects/tests/test_awssync.py

```diff
@@ -300,3 +300,167 @@ def mock_api(self, operation_name, kwarg):
+                        awssync.SyncData("email2@example.com", "project2", "Fall 2020"),
+                    ],
+                ),
+                awssync.Iteration(
+                    "Spring 2021",
+                    "98765",
+                    [
+                        awssync.SyncData("email3@example.com", "project3", "Fall 2021"),
+                        awssync.SyncData("email4@example.com", "project4", "Spring 2021"),
+                    ],
+                ),
+            ],
+        )
+
+        self.aws_tree3 = awssync.AWSTree(
+            "AWS Tree",
+            "12345",
+            [
+                awssync.Iteration(
+                    "Fall 2020",
+                    "54321",
+                    [
+                        awssync.SyncData("email1@example.com", "project1", "Fall 2020"),
+                        awssync.SyncData("email2@example.com", "project2", "Fall 2020"),
+                    ],
+                ),
+                awssync.Iteration(
+                    "Fall 2020",
+                    "98765",
+                    [
+                        awssync.SyncData("email3@example.com", "project3", "Fall 2021"),
+                        awssync.SyncData("email4@example.com", "project4", "Spring 2021"),
+                    ],
+                ),
+            ],
+        )
+
+    def test_repr_AWSTree(self):
+        self.assertEquals(str(self.awstree), "AWSTree('Name', '1234', [])")
+
+    def test_repr_Iteration(self):
+        self.assertEquals(str(self.iteration), "Iteration('Name', '1234', [])")
+
+    def test_repr_SyncData(self):
+        self.assertEquals(str(self.sync_data), "SyncData('email@example.com', 'Project X', 'Spring 2020')")
+
+    def test_awstree_to_syncdata_list(self):
+        self.assertEqual(self.aws_tree1.awstree_to_syncdata_list(), self.treelist)
+
+    def test_check_for_double_member_email(self):
+        # Test when there are no duplicate emails
+        self.assertFalse(self.sync.check_for_double_member_email(self.aws_list, self.sync_list))
+
+        # Test when there is a duplicate email
+        self.sync_list.append(awssync.SyncData("email4@example.com", "Spring 2022", "Project G"))
+        self.assertTrue(self.sync.check_for_double_member_email(self.aws_list, self.sync_list))
+
+    def test_check_current_ou_exists(self):
+        # Test when current semester OU does not exist
+        current_semester = "Fall 2022"
+        val1, val2 = self.sync.check_current_ou_exists(self.aws_tree1, current_semester)
+        self.assertEquals((val1, val2), (False, None))
+
+        # Test when current semester OU exists
+        current_semester = "Spring 2021"
+        val1, val2 = self.sync.check_current_ou_exists(self.aws_tree1, current_semester)
+        self.assertEqual((val1, val2), (True, "98765"))
+
+        # Test for code coverage, can't do it otherwise
+        val1, val2 = self.sync.check_current_ou_exists(self.aws_tree1)
+        self.assertTrue(True)
```

Comment:
Not every elegant, maybe use a MagicMock to mock the get_or_create_current_semester. Thjis will avoid assertTrue(True)